### PR TITLE
feat: Switcher - added node pool exclusions

### PR DESCRIPTION
**Pull request by mamari90**
### Type of changes

- [ ] Add new module
- [x] Update existing module
- [ ] Remove existing module
- [ ] Other

### Does this introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: azuredevops_build_definition_resource_switcher/aks_pipeline.tf [hunk 1]
```diff
@@ -125,6 +127,13 @@ resource "azuredevops_build_definition" "aks_pipeline" {
     allow_override = false
   }
 
+  variable {
+    name           = "TF_NODE_POOL_EXCLUSIONS"
+    value          = jsonencode(local.aks_config[count.index].exclusions)
+    is_secret      = false
+    allow_override = false
```
Comment (diegolagospagopa) [inline azuredevops_build_definition_resource_switcher/aks_pipeline.tf:134]: please add a description

File: azuredevops_build_definition_resource_switcher/aks_pipeline.tf [hunk 2]
```diff
@@ -18,6 +19,7 @@ locals {
         user_node_count        = cluster.user.nodes_on_stop
         system_node_count      = cluster.system.nodes_on_stop
         rg                     = cluster.rg
+        exclusions             = cluster.node_pool_exclusions == null ? [] : cluster.node_pool_exclusions
```
Comment (diegolagospagopa) [inline azuredevops_build_definition_resource_switcher/aks_pipeline.tf:22]: please add a description

**Review by diegolagospagopa — APPROVED** [review]

File: azuredevops_build_definition_resource_switcher/aks_pipeline.tf (hunk 1 above)
Comment (mamari90) [reply azuredevops_build_definition_resource_switcher/aks_pipeline.tf:146]: done

File: azuredevops_build_definition_resource_switcher/aks_pipeline.tf (hunk 2 above)
Comment (mamari90) [reply azuredevops_build_definition_resource_switcher/aks_pipeline.tf:22]: done

**Review by diegolagospagopa — APPROVED** [review]

**Pull request merged**